src/virtualship/cli/commands.py

```diff
@@ -45,9 +47,76 @@ def init(path):
     "path",
     type=click.Path(exists=True, file_okay=False, dir_okay=True, readable=True),
 )
-def fetch(path):
-    """Entrypoint for the tool."""
-    raise NotImplementedError("Not implemented yet.")
+def fetch(expedition_dir: str | Path) -> None:
+    """Entrypoint for the tool to download data based on area of interest."""
+    if isinstance(expedition_dir, str):
+        expedition_dir = Path(expedition_dir)
+
+    # Load schedule
+    schedule = _get_schedule(expedition_dir)
+    if schedule is None:
+        print("Error: Schedule file not found.")
+        return
+
+    # Extract area_of_interest details from the schedule
+    spatial_range = schedule.area_of_interest.spatial_range
+    time_range = schedule.area_of_interest.time_range
+    start_datetime = datetime.strptime(time_range.start_time, "%Y-%m-%d %H:%M:%S")
+    end_datetime = datetime.strptime(time_range.end_time, "%Y-%m-%d %H:%M:%S")
+
+    # Prompt for user credentials
+    username = input("username: ")
+    password = input("password: ")
+
+    # Define all datasets to download, including bathymetry
+    download_dict = {
+        "Bathymetry": {
+            "dataset_id": "cmems_mod_glo_phy_my_0.083deg_static",
+            "variables": ["deptho"],
+            "output_filename": "bathymetry.nc",
+            "force_dataset_part": "bathy",
+        },
+        "UVdata": {
+            "dataset_id": "cmems_mod_glo_phy-cur_anfc_0.083deg_PT6H-i",
+            "variables": ["uo", "vo"],
+            "output_filename": "default_uv.nc",
+        },
+        "Sdata": {
+            "dataset_id": "cmems_mod_glo_phy-so_anfc_0.083deg_PT6H-i",
+            "variables": ["so"],
+            "output_filename": "default_s.nc",
+        },
+        "Tdata": {
+            "dataset_id": "cmems_mod_glo_phy-thetao_anfc_0.083deg_PT6H-i",
+            "variables": ["thetao"],
+            "output_filename": "default_t.nc",
+        },
+    }
+
+    # Iterate over all datasets and download each based on area_of_interest
+    for dataset in download_dict.values():
+        copernicusmarine.subset(
+            dataset_id=dataset["dataset_id"],
+            variables=dataset["variables"],
+            minimum_longitude=spatial_range.min_longitude,
+            maximum_longitude=spatial_range.max_longitude,
+            minimum_latitude=spatial_range.min_latitude,
+            maximum_latitude=spatial_range.max_latitude,
+            start_datetime=start_datetime,
+            end_datetime=end_datetime,
+            minimum_depth=0.49402499198913574,
+            maximum_depth=5727.9169921875,
```

Comment:
Do we always need to download full-depth datasets? @ammedd, would there also be cases (when not using CTD?) that only upper/surface ocean data are needed? Or would that just overcomplicate the script?

Reply:
We can add this very easily to the `area_of_interest` . Do we have checks to see if the data covers the survey?

Reply:
The are indeed (many) cases where the full-depth is not needed. It would be good to check the ship_config for the maximum depth needed.